src/main.rs

```diff
@@ -303,18 +301,26 @@ fn main() -> Result<()> {
     let (pkg_set, resolve) = resolve_ws(&workspace)?;
 
     let mut patched = false;
+
+    let mut handle_custom_metadata =
```

Comment:
If possible, I would prefer this to be in its own function

Reply:
Sure, we'll just have to pass the resolver as an argument it seems; the way it's done in here is quite a bit cleaner.
I think I have an idea on how to recompose it even better though... Shall I try?

Reply:
Please go ahead. Looking forward to merge.